constraints/match.js

```diff
@@ -124,7 +126,7 @@ function matchSP(inputTree, pTree, sCat, options)
 	*  - either it is lexical (sParent.func is false) OR requireLexical is false
 	*  - either it has an overt head (sParent.silent is false) OR requireOvertHead is false
 	*/
-	if(sParent.cat === sCat
+	if(((sParent.cat === sCat ) && !options.anySyntacticConstituent)
```

Comment:
I think the logic here actually needs to be:
if(((sParent.cat === sCat ) || options.anySyntacticConstituent) ...
because if we're satisfied with any syntactic category, then it's also not a problem if it is some specified category. 
Another thing to watch out for: what if anySyntacticConstituent === true and then sCat isn't defined? 
maybe best to do the opposite order of conditions to handle that: 
if((options.anySyntacticConstituent || (sParent.cat === sCat )) ...